### PR TITLE
Tests: Use separate schemas for sharded and unsharded

### PR DESCRIPTION
**Pull request by bquorning**
Using the same schema for sharded and unsharded databases risks us having falsely positive tests.

### CODE REVIEW

**Comment by bquorning** [comment]
Migrations are still failing. Someone please have a look.

File: test/connection_switching_test.rb [hunk 1]
```diff
@@ -616,17 +616,6 @@ class ShardedModel < ActiveRecord::Base
...
-        it "supports implicit joins" do
-          accounts = Account.includes(:tickets)
-          accounts = accounts.references(:tickets) if ActiveRecord::VERSION::MAJOR >= 4
-          assert_equal ["slave_name", "slave_name2"], accounts.order('tickets.id').map(&:name).sort
-        end
-
-        it "supports joins" do
-          accounts = Account.joins('LEFT OUTER JOIN tickets ON tickets.account_id = accounts.id').map(&:name).sort
-          assert_equal ["slave_name", "slave_name2"], accounts
-        end
```
Comment (grosser) [inline test/connection_switching_test.rb]: need a replacement test for this that is not cross-shard ?

Comment (bquorning) [reply test/connection_switching_test.rb]: Done. We should probably test joins between sharded tables as well as between unsharded tables?

**Comment by grosser** [comment]
joins between sharded/unsharded should blow up ?

On Fri, Sep 22, 2017 at 9:51 AM, Benjamin Quorning <notifications@github.com

**Comment by bquorning** [comment]
> joins between sharded/unsharded should blow up ?

Yes. Do you mean that we should explicitly test for it?

**Comment by grosser** [comment]
I'd prefer to test edge-case behavior just so we know it's not some weird
MethodMissing error ...

On Mon, Sep 25, 2017 at 4:04 AM, Benjamin Quorning <notifications@github.com

**Comment by grosser** [comment]
was only missing a `connect` at the start :D

**Comment by bquorning** [comment]
Green build! All good?
